src/Cms/ModelWithContent.php

```diff
@@ -129,14 +129,25 @@ abstract protected function commit(
 	 */
 	public function content(string|null $languageCode = null): Content
 	{
+		// get the targeted language
+		$language = Language::ensure($languageCode);
+
+		// fetch a specific version in preview render mode
+		// @todo this entire block can be radically simplified as soon
+		// as the models use the versions exclusively.
+		if (VersionId::$render ?? null) {
+			$version = $this->version(VersionId::render($this));
+
+			if ($version->exists($language) === true) {
+				return $version->content($language);
+			}
```

Comment:
`Understanding`: What happens if the render mode is set to changes, but the currently requested language doesn't exist? Shouldn't it still show changes, but in the default language? I think based on this code it would fall back to the published version instead.

Reply:
Uff, it's creating another brain knot for me. Let me try to break it down. 

- The English version has changes
- I'm switching to German 
- German already has published translations for all fields 
- I want to make some additional changes there as well. 
- The changes version wouldn't exist at this point. 

But wouldn't it be completely misleading or even wrong to fall back to the changes in English here? That's at least how I think about it.

Reply:
I think it's actually fine how it is.